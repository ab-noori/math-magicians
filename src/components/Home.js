import algorithm from '../assets/algorithm.gif';

function Home() {
  return (
    <div className="home">
      <div className="intro-frame">
        <h2>Welcome to our page!</h2>
        <p className="text">
          <b>Math magicians</b>
          {' '}
          is a website for all fans of mathematics.
          <br />
          It is a Single Page App (SPA) that allows users to:
        </p>
        <ul>
          <li className="listItem">Make simple calculations.</li>
          <li className="listItem">Read a random design-related quote.</li>
        </ul>
        <p>
          The app is developed using
          {' '}
          <i>React,</i>
          {' '}
          and styled through the
          {' '}
          <i>React basic styling (pre-processed CSS file).</i>
          {' '}
          Navigation is made possible through
          {' '}
          <i>React Router.</i>
        </p>
      </div>

      <div className="algo-frame">
        <img src={algorithm} alt="algoritm" />
      </div>
    </div>
  );
}
export default Home;
