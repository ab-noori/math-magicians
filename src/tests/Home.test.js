import React from 'react';
import renderer from 'react-test-renderer';
import Home from '../components/Home';

describe('Home component', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Home />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
