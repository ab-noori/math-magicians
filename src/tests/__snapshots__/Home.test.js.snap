// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component renders correctly 1`] = `
<div
  className="home"
>
  <div
    className="intro-frame"
  >
    <h2>
      Welcome to our page!
    </h2>
    <p
      className="text"
    >
      <b>
        Math magicians
      </b>
       
      is a website for all fans of mathematics.
      <br />
      It is a Single Page App (SPA) that allows users to:
    </p>
    <ul>
      <li
        className="listItem"
      >
        Make simple calculations.
      </li>
      <li
        className="listItem"
      >
        Read a random design-related quote.
      </li>
    </ul>
    <p>
      The app is developed using
       
      <i>
        React,
      </i>
       
      and styled through the
       
      <i>
        React basic styling (pre-processed CSS file).
      </i>
       
      Navigation is made possible through
       
      <i>
        React Router.
      </i>
    </p>
  </div>
  <div
    className="algo-frame"
  >
    <img
      alt="algoritm"
      src="algorithm.gif"
    />
  </div>
</div>
`;
